test/gtest/mha_find20.cpp

```diff
@@ -0,0 +1,395 @@
+            {
+                double error = miopen::rms_range(outputTensorResults[id]->tensorFloat.data,
+                                                 tensors[id]->tensorFloat.data);
+                const double tolerance = 1e-3;
+
+                EXPECT_TRUE(std::isfinite(error) && error <= tolerance)
+                    << "Outputs do not match each other. Error:" << error;
+            }
+        }
+
+        std::cerr << "Finished testing forward solution functions." << std::endl;
+    }
+
+    void TestRunSolutionsBackward(Handle& handle, const std::vector<miopenSolution_t>& solutions)
+    {
+        std::cerr << "Testing backward solution functions..." << std::endl;
+
+        std::cerr << "Finished testing backward solution functions." << std::endl;
+    }
+
+    void Finalize() { EXPECT_EQUAL(miopenDestroyProblem(problem), miopenStatusSuccess); }
+
+private:
+    bool IsDropoutTensorId(miopenTensorArgumentId_t id) const
+    {
+        return id == miopenTensorMhaDropoutProbability || id == miopenTensorMhaDropoutSeed ||
+               id == miopenTensorMhaDropoutOffset;
+    }
+
+    void CreateTensor(miopenTensorArgumentId_t id,
+                      unsigned int n = 1,
+                      unsigned int c = 1,
+                      unsigned int h = 1,
+                      unsigned int w = 1,
+                      bool generate  = true)
+    {
+        // TODO Unused for now
+        bool floatFlag = !IsDropoutTensorId(id);
+        tensors[id]    = TensorStructPtr(new TensorStruct(floatFlag));
+
+        tensors[id]->tensorFloat = tensor<float>{test_n, test_c, test_h, test_w};
+
+        if(generate)
+        {
+            tensors[id]->tensorFloat.generate(tensor_elem_gen_integer{17});
+        }
+
+        EXPECT_EQUAL(miopenSetProblemTensorDescriptor(problem, id, &tensors[id]->tensorFloat.desc),
+                     miopenStatusSuccess);
+    }
+
+    void Initialize()
+    {
+        Mha_descriptor.SetParams(scale);
+
+        EXPECT_EQUAL(
+            miopenCreateMhaProblem(&problem, &Mha_descriptor, miopenProblemDirectionForward),
+            miopenStatusSuccess);
+
+        if(isForward)
+        {
+            CreateTensor(miopenTensorMhaK, test_n, test_c, test_h, test_w);
+            CreateTensor(miopenTensorMhaQ, test_n, test_c, test_h, test_w);
+            CreateTensor(miopenTensorMhaV, test_n, test_c, test_h, test_w);
+
+            CreateTensor(miopenTensorMhaDescaleK);
+            CreateTensor(miopenTensorMhaDescaleQ);
+            CreateTensor(miopenTensorMhaDescaleV);
+            CreateTensor(miopenTensorMhaDescaleS);
+            CreateTensor(miopenTensorMhaScaleS);
+            CreateTensor(miopenTensorMhaScaleO);
+
+            CreateTensor(miopenTensorMhaDropoutProbability);
+            CreateTensor(miopenTensorMhaDropoutSeed);
+            CreateTensor(miopenTensorMhaDropoutOffset);
+
+            CreateTensor(miopenTensorMhaO, test_n, test_c, test_h, test_w);
```

Comment:
```suggestion
            CreateTensor(miopenTensorMhaO, test_n, test_c, test_h, test_w, false);
```
This is also an output tensor, like all the following.